js/treeselect_filter.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @file
+ * Javascript behaviour for the treeselect filter.
+ */
+
+(function ($, Drupal) {
+
+  'use strict';
+
+  /**
+   * Attaches treeselect filter for term reference fields.
+   */
+  Drupal.behaviors.treeselectFilter = {
+    attach: function (context, settings) {
+      $('.treeselect_filter', context).once('initTreeselect').each(function (i, el) {
+        el = $(el);
+        var field = el.attr('id').split('-')[1];
+        var data = settings.contentpoolClient.filterData[field];
+        var elementId = '#treeselect_filter-' + field;
+
+        Vue.component('treeselect', VueTreeselect.Treeselect);
+
+        new Vue({
+          el: elementId,
+          data: data,
+          watch: {
+            value: function(newValue, oldValue) {
+              // Push values into the form field.
+              $('input[name="filter-' + field + '"]').val(newValue.toString());
```

Comment:
That seems strange, usually this is done via v-model.
I think you should just make sure the hidden input is part of your vue app also and then use v-model "value" on the hidden input also. You can just work with the value as usual in vue, e.g. see https://vue-treeselect.js.org/#basic-features